subsys/net/ip/net_context.c

```diff
@@ -133,24 +134,32 @@ int net_context_get(sa_family_t family,
 	}
 #endif
 
-	if (IS_ENABLED(CONFIG_NET_SOCKETS_PACKET)) {
-		if (type == SOCK_RAW) {
+	if (!IS_ENABLED(CONFIG_NET_SOCKETS_CAN) && family == AF_CAN) {
+		NET_ASSERT_INFO((family != AF_CAN), "AF_CAN disabled");
+		return -EPFNOSUPPORT;
+	}
+
+	if (type == SOCK_RAW) {
+		if (IS_ENABLED(CONFIG_NET_SOCKETS_PACKET) &&
+		    IS_ENABLED(CONFIG_NET_SOCKETS_CAN)) {
+			if (family != AF_PACKET && family != AF_CAN) {
+				return -EINVAL;
+			}
+		} else if (IS_ENABLED(CONFIG_NET_SOCKETS_RAW) &&
```

Comment:
@jukkar: Another issue with this patch. As @galak points out, CONFIG_NET_SOCKETS_RAW is dead symbol. Well, I don't understand this "if" staircase which you with @rveerama1 added here, so I hope you guys can look into what would be the proper fix.